### ISSUE TITLE
Affine transform between torchio.ScalarImage and nibabel.load do not match

### ISSUE THREAD

**Issue by rayryeng** [fepegar/torchio#469]
**🐛Bug**
I am using [the following NIfTI volume](https://www.dropbox.com/s/3n0b9e6sx81azrd/img0009.nii.gz?dl=1) that is part of a dataset that I am using (don't worry, this is public data and not confidential).  When you use `torchio.ScalarImage` and examine the affine transform after it has been loaded, it is not the same as if you were to use `nibabel.load`.  This causes havoc when I use the `ToCanonical`transform because the volume is not oriented correctly.

**To reproduce**

Please download the file from the link above.  Once you do, do the following:

```python
import torchio as tio
import nibabel as nib

vol_path = 'img0009.nii.gz'
img_s = tio.ScalarImage(vol_path)
print(img_s.affine)
# array([[-0.97656202,  0.        ,  0.        ,  0.        ],
#       [ 0.        , -0.97656202,  0.        ,  0.        ],
#       [ 0.        ,  0.        ,  2.5       ,  0.        ],
#       [ 0.        ,  0.        ,  0.        ,  1.        ]])
img_n = nib.load(vol_path)
print(img_n.affine)
# array([[  -0.97656202,    0.        ,    0.        ,  249.51159692],
#       [   0.        ,    0.97656202,    0.        , -249.51159692],
#       [   0.        ,    0.        ,    2.5       , -185.        ],
#       [   0.        ,    0.        ,    0.        ,    1.        ]])
```

I am using Nibabel version 3.2.1 and torchio version 0.18.29.

**Expected behavior**

I would expect the affine transforms between the calls to `torchio.ScalarImage` and `nib.load` to be exactly the same.  Why is the affine transform modified?

**Actual behavior**

The affine transforms are different.

**System info**

```
Platform: Linux-4.15.0-91-generic-x86_64-with-debian-buster-sid
TorchIO:  0.18.29
PyTorch:  1.7.1
NumPy:    1.19.2
Python:   3.7.3 (default, Mar 27 2019, 22:11:17)
[GCC 7.3.0]
```

**Comment by fepegar** [comment]
Hi, @rayryeng. The NIfTI header of that image is not valid: both the qform and sform codes are 0, i.e., unknown, therefore NiBabel (or SimpleITK, used by default in TorchIO) need to figure out somehow a fall-back affine. More info in the [NiBabel docs](https://nipy.org/nibabel/nifti_images.html#the-fall-back-header-affine).

Slicer, which uses SimpleITK, shows an image flipped in the antero-posterior axis:
![Screenshot](https://user-images.githubusercontent.com/12688084/108206554-745ad700-711e-11eb-9656-5eddb8b24f24.png)

How was this file created?

**Comment by rayryeng** [comment]
Ahh I understand.  Thanks!  This is `img0009.nii.gz` that is part of the training split from the Beyond the Cranial Vault Abdomen Challenge: https://www.synapse.org/#!Synapse:syn3193805/wiki/217789.  Since `ToCanonical` relies on the affine transform, do you have a suggestion on how to mitigate this shortcoming in the data?

**Comment by fepegar** [comment]
> This is img0009.nii.gz that is part of the training split from the Beyond the Cranial Vault Abdomen Challenge:

The authors should fix their dataset! I wonder how they performed the conversion.

> Since ToCanonical relies on the affine transform, do you have a suggestion on how to mitigate this shortcoming in the data?

I suppose that, assuming all images in the dataset are all oriented in the same way, you can just stop using `ToCanonical`. Or you can try to fix the dataset headers, modifying maybe the qform_code.

**Comment by rayryeng** [comment]
Thanks for the insight! I suppose I can abandon the use of `ToCanonical` for now. If both TorchIO and NiBabel use the same backend, why would the affine transforms be different between the platforms? This is my last question. Otherwise I'll close this issue as it's beyond your contro

**Comment by fepegar** [comment]
No worries! Actually, TorchIO uses NiBabel only if SimpleITK fails to read the file.

**Comment by rayryeng** [comment]
I got it! Thank you so much! I shall close this.
